tests/integration/test_workflow_yaml_validation.py

```
@@ -0,0 +1,500 @@
+                            f"Invalid PR activity type: {activity_type}"
+    
+    def test_push_trigger_has_branches(self, workflow_content):
+        """Push triggers should specify branches when appropriate."""
+        on_trigger = workflow_content.get('on', {})
+        
+        if isinstance(on_trigger, dict) and 'push' in on_trigger:
+            push_config = on_trigger['push']
+            
+            # If it's a dict, it should have branches or paths
+            if isinstance(push_config, dict):
+                has_filter = 'branches' in push_config or 'paths' in push_config
```

Ruff (0.14.6)

Local variable `has_filter` is assigned to but never used

Remove assignment to unused variable `has_filter`

(F841)